src/otx/algo/classification/otx_dino_v2.py

```diff
@@ -65,8 +63,8 @@ def forward(self, imgs: torch.Tensor, labels: torch.Tensor = None) -> torch.Tens
 class DINOv2RegisterClassifier(OTXMulticlassClsModel):
     """DINO-v2 Classification Model with register."""
 
-    def __init__(self, num_classes: int, config: DictConfig) -> None:
-        self.config = config
+    def __init__(self, num_classes: int, config: DictConfig | dict) -> None:
```

Comment:
What's the reason behind of changing so many typing from `DictConfig` to `DictConfig | dict` not just for this line but many others?

Reply:
When `parse_args` in `argparser`, config is instantiated as a nested dictionary, so I modified it to accept a `dict` type.
Actually, it would be fine to modify it just for that class (the class that uses a dict argument named config in the config file// + OVModel), but for now I did the same for the model class, which uses config in the same way.

Reply:
Can we choose only one of either? If it is not an inevitable situation, we'd better to avoid multiple typings as possible to be developer or user friendly. In this case, it is even more so because they are used for exactly the same purpose.

> When parse_args in argparser, config is instantiated as a nested dictionary, so I modified it to accept a dict type

In addition, do we have custom resolvers for `DictConfig`? Are those correctly parsed?

Reply:
The resolver for `DictConfig` is the same as before. (otx.core.config._init_.register_configs)
Unfortunately, jsonargparse does not provide `DictConfig` as a default parser.
(use dict by default.) -> Let's see if there's a way to do this.

So I guess there are 3 ways to do it.
Let's say we leave the configs of the mm-related models as they are, since they use only `DictConfig` (Currently they use `read_mmconfig`, so we don't need to change that.). 

The models that currently take a `DictConfig` as an argument that should be modified are `Dinov2` & `OVModel` for now.

1. leave this as `DictConfig` and specify `DictConfig` in the configuration file.
```
model:
  class_path: otx.algo.classification.otx_dino_v2.DINOv2RegisterClassifier
  init_args:
    num_classes: 1000
    config:
      class_path: omegaconf.dictconfig.DictConfig
      init_args:
        content:
          backbone:
            name: dinov2_vits14_reg
            frozen: false
          head:
            in_channels: 384
            num_classes: 1000
```
2. change this as `dict`
3. Modifying arguments explicitly, without using config(dict, DictConfig)

For example 3, OVModel could be changed to something like this.
```python
class OVModel(OTXModel, Generic[T_OTXBatchDataEntity, T_OTXBatchPredEntity]):
    def __init__(self, num_classes: int, model_name:  str, ...) -> None:
```

Which method do you think is best?

Or maybe we should just make sure we get all of the `DictConfig | dict` and then figure out some other good way to do it.

Reply:
1. It should be the option 1. This is because they were originally `MMConfig` but we integrated them into our `DictConfig` format. However, we did some refactoring to separate them into `algo/<task>/mmconfigs/*.yaml`, so that there is no need to keep them as `DictConfig` format. Finally, we will change again from `DictConfig` to `MMConfig` to get back them as they were. Not only here but also please revert the typing in `MM*CompatibleModel` such as
https://github.com/openvinotoolkit/training_extensions/blob/5d60742dcea63d49ad5da05deb7bbf0254bb8b4b/src/otx/core/model/entity/action_detection.py#L35

2. My second question is about the custom resolvers. How `jsonargparse` can interpret the custom Python object from the config file? The most important usage for this is
```yaml
    - _target_: torchvision.transforms.v2.ToDtype
      dtype: ${as_torch_dtype:torch.float32}
      scale: True
```
To interpret `torch.float32`, we introduced the custom resolver. I could see there remains this configuration only in this PR without the `torch.dtype` resolving:
https://github.com/openvinotoolkit/training_extensions/blob/c7ddf78d5460aee1a3095957cff14f83e7e1fa54/src/otx/recipe/_base_/data/torchvision_base.yaml#L7-L13

However, supporting TorchVision v2 transform is important. I'd like to ask you to add a test instantiating `OTXDataModule` including the `torch.dtype` resolving from the configuration file. It is no matter on how to do this. If there is a native support for this with `jsonargparse`, you can choose it, but the important thing is that we should support TorchVision v2 transform.

Reply:
Thanks for the insight.
I proceeded with #1 as we discussed. - https://github.com/openvinotoolkit/training_extensions/pull/2805/commits/13b078582386d3562954b4e7d9aac5e3ed9514cb

I also checked out question 2. 
```
    - class_path: torchvision.transforms.v2.ToDtype
      init_args:
        dtype: ${as_torch_dtype:torch.float32}
        scale: True
```
We can use the transform above.